### PR TITLE
fix(model): Use the correct PURL type for CocoaPods

### PR DESCRIPTION
**Pull request by ksg97031**
According to the purl-spec and ossindex, the name of CocoaPods is "CocoaPods" not "Pod".

[1] https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#cocoapods
[2] https://ossindex.sonatype.org/ecosystem/cocoapods

### CODE REVIEW

**Review by sschuberth — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please use `fix` as the Conventional Commit type, like `fix(model): Use the correct PURL type for CocoaPods`.

**Comment by sschuberth** [comment]
> Thanks for your contribution! Please use `fix` as the Conventional Commit type, like `fix(model): Use the correct PURL type for CocoaPods`.

Also footers that refer to URLs should be referred to from the body text, and the foot notes should have a colon after the square brackets. See e.g. 4f9609e979ddfe1422e6241a1a1b201f510eb781.

File: plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt [hunk 1]
```diff
@@ -253,7 +253,7 @@ private fun getPackageReferences(podfileLock: File): Set<PackageReference> {
 
     fun createPackageReference(name: String): PackageReference =
         PackageReference(
-            id = Identifier("Pod", "", name, versionForName.getValue(name)),
+            id = Identifier("CocoaPods", "", name, versionForName.getValue(name)),
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt:256]: This actually needs to be reverted and *only* the change in the other file be kept.

File: model/src/main/kotlin/utils/Extensions.kt [hunk 2]
```diff
@@ -208,6 +208,7 @@ fun Identifier.getPurlType() =
         "npm" -> PurlType.NPM
         "nuget" -> PurlType.NUGET
         "pypi" -> PurlType.PYPI
+        "cocoapods" -> PurlType.COCOAPODS
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/Extensions.kt:211]: This needs to map `"pod"` (lowercase name of ORT's internal type) to `PurlType.COCOAPODS`. And please sort this in alphabetically before the `"pypi"` line.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Comment by sschuberth** [comment]
> According to the [1] and [2], the name of CocoaPods is "CocoaPods" not "Pod".

Also, please amend your commit and hard-wrap this line after 75 chars.

File: model/src/main/kotlin/utils/Extensions.kt (hunk 2 above)
Comment (ksg97031) [reply model/src/main/kotlin/utils/Extensions.kt:211]: I'm sorry, I just understood.

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
Good point about the test fixes. Please also fix the occurrences in [plugins/package-managers/cocoapods/src/funTest/assets/projects/synthetic/dep-tree-expected-output.yml](https://github.com/oss-review-toolkit/ort/blob/main/plugins/package-managers/cocoapods/src/funTest/assets/projects/synthetic/dep-tree-expected-output.yml) and simply amend your first commit with all the changes to tests.

**Comment by sschuberth** [comment]
Please don't put merge commits into PRs, but rebase instead.

**Comment by ksg97031** [comment]
> Please don't put merge commits into PRs, but rebase instead.

I've rebased the commit to avoid merge commits. 
I apologize for any inconvenience this may cause.

**Comment by sschuberth** [comment]
> I apologize for any inconvenience this may cause.

No need to apologize. Thanks for addressing all review comments so quickly!

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite the unrelated `OsvFunTest` failure.

**Pull request merged**